Azure/azure-cli
# [App Service] `az functionapp create`: Remove unnecessary app settings for flex function apps

**Related command**
`az functionapp create`

**Description**
- The backend logic has blocked the creation of flex function apps with the FUNCTIONS_EXTENSION_VERSION app setting. Therefore, we need to make sure that this app setting is not added in the creation flow. 
 - Additionally, we should not be setting distributed tracing on by default for flex function apps. 
 
**Testing Guide**
- `az functionapp create` for a flex app no longer adds the FUNCTIONS_EXTENSION_VERSION app setting
- `az functionapp create` for a java flex app no longer adds the APPLICATIONINSIGHTS_ENABLE_AGENT app setting

**History Notes**
- [AppService] `az functionapp create`: Remove FUNCTIONS_EXTENSION_VERSION app setting for flex function apps
- [AppService] `az functionapp create`: Remove APPLICATIONINSIGHTS_ENABLE_AGENT app setting for flex function apps
- [AppService] `az functionapp create`: Remove AzureWebJobsDashboard app setting for flex function apps
- [AppService] `az functionapp create` and `az functionapp list-flexconsumption-locations`: Make request to geo regions API
- [AppService] `az functionapp create` and `az functionapp vnet-integration add`: Register subscription to Microsoft.App RP for flex function app vnet integration
- [AppService] `az functionapp create`: Delete newly created resources specific to flex if the app creation fails

---

This checklist is used to make sure that common guidelines for a pull request are followed.
